packages/@apphosting/adapter-nextjs/e2e/app.spec.ts

```diff
@@ -114,4 +124,30 @@ describe("app", () => {
       "private, no-cache, no-store, max-age=0, must-revalidate",
     );
   });
+
+  it("should have x-fah-adapter header and no x-fah-middleware header on all routes", async () => {
```

Comment:
can we manually codegen a middleware.ts in the test project that affects a couple of these routes?

Reply:
Updated the `run-local.ts` file to support this. Would be great to have our e2e tests running against different next versions. I can look at that after this.